azure-iot-hub/azure/iot/hub/iothub_registry_manager.py

```diff
@@ -56,29 +56,51 @@ class IoTHubRegistryManager(object):
     based on top of the auto generated IotHub REST APIs
     """
 
-    def __init__(self, connection_string):
+    def __init__(self, connection_string=None, host=None, auth=None):
         """Initializer for a Registry Manager Service client.
 
         After a successful creation the class has been authenticated with IoTHub and
         it is ready to call the member APIs to communicate with IoTHub.
 
         :param str connection_string: The IoTHub connection string used to authenticate connection
-            with IoTHub.
+            with IoTHub if we are using connection_str authentication. Default value: None
+        :param str host: The Azure service url if we are using token credential authentication.
+            Default value: None
+        :param str auth: The Azure authentication object if we are using token credential authentication.
+            Default value: None
 
         :returns: Instance of the IoTHubRegistryManager object.
         :rtype: :class:`azure.iot.hub.IoTHubRegistryManager`
         """
-        self.auth = ConnectionStringAuthentication(connection_string)
-        self.protocol = protocol_client(self.auth, "https://" + self.auth["HostName"])
-        self.amqp_svc_client = iothub_amqp_client(
-            self.auth["HostName"], self.auth["SharedAccessKeyName"], self.auth["SharedAccessKey"]
-        )
+        if connection_string:
+            self.auth = ConnectionStringAuthentication(connection_string)
+            self.protocol = protocol_client(self.auth, "https://" + self.auth["HostName"])
+            self.amqp_svc_client = iothub_amqp_client(
+                self.auth["HostName"],
+                self.auth["SharedAccessKeyName"],
+                self.auth["SharedAccessKey"],
+            )
+        else:
+            self.auth = auth
+            self.protocol = protocol_client(self.auth, "https://" + host)
+            self.amqp_svc_client = None
+
+    @classmethod
+    def from_connection_string(cls, connection_string):
```

Comment:
>from_connection_string [](http://example.com/codeflow?start=8&length=22)

from_connection_string and from_token_credential need documentation strings.

Reply:
Added